88 create jupyter book

phchen5 opened the pull request:
create jupyter book
edit notebook to remove code
fix scripts that are outputting wrong stuff

phchen5 committed:
debug and create jupyterbook

phchen5 committed:
hide import code cell

phchen5 committed:
edit notebook

phchen5 committed:
edit prediction_plot script to include original_plot

phchen5 committed:
edit notebook more

phchen5 committed:
Merge branch 'main' into 88-create-jupyter-book
merge into branch

phchen5 committed:
edit eda to fix description output

phchen5 committed:
done with template of jupyterbook

SoloSynth1 committed:
fix JB warnings on headers; remove unused environment setup instructions

SoloSynth1 committed:
fix incorrect bibtex citing

SoloSynth1 committed:
refactoring
- remove intermediate files in `report/_build`
- add rendered HTML in `docs/`
- fix incorrect book title in `report/_config.yml`
- lint bibtex's format

SoloSynth1 committed:
refactoring
- remove intermediate files in `report/_build`
- add rendered HTML in `docs/`
- fix incorrect book title in `report/_config.yml`
- lint bibtex's format

SoloSynth1 committed:
add `.nojekyll` to bypass Jekyll processing

SoloSynth1 reviewed (changes requested):
Changes are needed as there are errors during the jupyter book building process:

![image](https://github.com/UBC-MDS/project-avalon/assets/17952490/99a83e56-43da-4169-b316-f57e54906ecf)

Blocked and probably will be fixed by #68.

Base on the above reason this PR is on hold.

SoloSynth1 commented:
Linking #93 as the rendered HTML files would be ready to host once this PR merges to `main`.

phchen5 committed:
edit text in notebook intro

phchen5 committed:
edit missing values script to include variables

phchen5 committed:
edit eda text in notebook

phchen5 committed:
change col name of index in correlations

phchen5 committed:
changed all text and analysis with reference to figures and values

phchen5 committed:
small modification on referencing

phchen5 committed:
fix referencing to row value

phchen5 committed:
edit figure captions

phchen5 committed:
edit script to add help argument

SoloSynth1 committed:
lint script files formatting

SoloSynth1 committed:
Merge remote-tracking branch 'origin/88-create-jupyter-book' into 88-create-jupyter-book

# Conflicts:
#	scripts/autocorrelation_plot.py
#	scripts/modelling.py
#	scripts/prediction_plot.py

SoloSynth1 committed:
lint script files formatting

SoloSynth1 committed:
add get_metrics.py and intermediate csv file

SoloSynth1 committed:
glue-ify notebook
- change the notebook to use glue for observations
- add `index_col` in `read_csv` to set index
- re-render `docs/`
- update `README.md` on steps to build the jupyter book site

SoloSynth1 committed:
add glue text for ARIMA MAE results in notebook

SoloSynth1 committed:
add jupyter-book config
- add settings for HTML and repository in `_config.yml`
- fix incorrect glue reference for `arime_mae`
- re-render jupyter book files

SoloSynth1 committed:
change to use `click.option` instead of `click.argument`

SoloSynth1 committed:
add logo to the jupyter book

SoloSynth1 reviewed (approved):
Ben's part looks good for me

@phchen5 @zywkloo @MoNorouzi23 If you want to take a glance on my changes feel free to do so

If no questions I will merge it into `main`.

SoloSynth1 committed:
Merge branch 'main' into 88-create-jupyter-book
